src/route/dataset-route.ts

```diff
@@ -73,9 +84,10 @@ apiRoute.post('/', upload.single('csv'), async (req: Request, res: Response) =>
         res.json(err);
         return;
     }
-    const dataset = Dataset.createDataset(internalName, 'BetaUser');
+    const dataset = Dataset.createDataset(title, 'BetaUser');
     const saved_dataset_record = await dataset.save();
-    const uploadDTO = await uploadCSV(req.file?.buffer, saved_dataset_record);
+    saved_dataset_record.addTitleByString(title, lang);
+    const uploadDTO = await uploadCSVToBlobStorage(req.file?.buffer, saved_dataset_record);
```

Comment:
When we receive file data via HTTP POST it gets encoded in the Request Body.
In these cases we should be able to tell the HTTP Server implementation to only read the Request Header and then pass the body to us as an open file descriptor. Then we can decode the parameters one-by-one and when it gets to the (potentially large) file data, pass the FD (or a derivative) onto lower layers so that we can process the data page-by-page straight from the incoming TCP socket to the outgoing buffer (disk or TCP).
Then we don't have to worry about how much data the user sends us other than to check the parameter size and ensure we have enough space in the backing store. i.e. we don't need more than a few pages (32K?) of RAM for any sized upload.

Reply:
I'm not 100% convinced we can do this using NodeJS and the layers of firewall, etc.  If you can provide me an example which proves me wrong I am happy to give it a look.  Ultimately NodeJS is its own Web Server with ExpressJS providing the framework for handling the post request.

Reply:
I'm not sure about your API but other HTTP Request/Response APIs often provide a port object that can be used to read the body and there's a way to tell the framework to not process it.

https://expressjs.com/en/5x/api.html points me at http://expressjs.com/id/resources/middleware/body-parser.html which says it doesn't do multipart bodies and suggests some other modules.

Picking busboy (the first one) https://www.npmjs.com/package/busboy#readme, there's an example (the second one) and documentation: "file(< string >name, < Readable >stream, < object >info) - Emitted for each new file found. name contains the form field name. stream is a Readable stream containing the file's data."

...that shows how to parse multipart/form-data POST bodies (which is the only encoding that supports file uploads from browsers via HTML form submission).

Which module are we using for processing multipart/form-data?